### ISSUE TITLE
Some problems about restoration_video_inference.py

### ISSUE THREAD

**Issue by Shuweis** [open-mmlab/mmediting#439]
When I run the restoration_video_inference.py like:
python demo/restoration_video_demo.py ./configs/restorers/edvr/edvrm_wotsa_x4_g8_600k_reds.py https://download.openmmlab.com/mmediting/restorers/edvr/edvrm_wotsa_x4_8x4_600k_reds_20200522-0570e567.pth /home/ssw20/mmediting/data/REDS/train_sharp_bicubic/X4/000/ ./outputs/calendar_EDVR --start_idx=0 --window_size=5

Some problems happened: TypeError: restoration_video_inference() takes 4 positional arguments but 5 were given

I cannot find what happened.
Hope for your reply!

**Comment by ckkelvinchan** [comment]
Did you use the latest version of MMEditing? In particular, you should see 
```
def restoration_video_inference(model, img_dir, window_size, start_idx, filename_tmpl):
```
in [restoration_video_inference.py](https://github.com/open-mmlab/mmediting/blob/966eeb3e8d9c00da2433cc17b6b8463fcbb8e8d0/mmedit/apis/restoration_video_inference.py#L21)

**Comment by Shuweis** [comment]
Yes, I run the latest version of MMEditing and the error happened

**Comment by ckkelvinchan** [comment]
> Yes, I run the latest version of MMEditing and the error happened

Could you take a screenshot of `restoration_video_inference.py`, and provide me with the complete error log? I have just tried on my side and there is no problem.

**Comment by Shuweis** [comment]
Sorry,I forgot to reply,I double-checked it carefully and found that it was caused by a mismatch between the environment and the file version.

**Comment by Shuweis** [comment]
I got another question.When I attempted to print the params.grad of basicvsr model in restoration_video_demo.py,I got:

-->name: generator.forward_resblocks.main.2.29.conv2.bias -->grad_requirs: True  -->grad_value: None
-->name: generator.fusion.weight -->grad_requirs: True  -->grad_value: None
-->name: generator.fusion.bias -->grad_requirs: True  -->grad_value: None
-->name: generator.upsample1.upsample_conv.weight -->grad_requirs: True  -->grad_value: None
I do not know why all the params.grad is None.

**Comment by ckkelvinchan** [comment]
As only inference is done in the demo (i.e. there is no any `.backward()`), it is normal that the gradient is `None`.

**Comment by Shuweis** [comment]
However,I use edvrm model to do the same pipeline, its grad_value is normal（not None）.

**Comment by Shuweis** [comment]
I try to remove torch.no_grad(),and run the demo.

result = model(lq=_add_batch_one(img_tensor).to(device), test_mode=True)['output'].to(device) 
print('result.requires_grad: ',result.requires_grad)

When I run the edvrm,result.requires_grad is True.However, When I run the basicvsr,the result is None and it cannot use the backward operation. I cannot find what happened. Hope for your reply.

**Comment by ckkelvinchan** [comment]
Since `with torch.no_grad()` is used in [BasicVSR()](https://github.com/open-mmlab/mmediting/blob/master/mmedit/models/restorers/basicvsr.py#L154). 

Nevertheless, it does not affect the inference performance.

**Comment by Shuweis** [comment]
Okay,I got that. If compute the grad for the basicvsr in inference period, I found it is hard for the model to be run on single GPU.Thank you for your reply!

**Comment by ckkelvinchan** [comment]
The demo is used for inference only. For training, please use `tools/dist_train.sh` or `tools/train.py`
